### DISCUSSION TITLE
hydrate the entire document make it too easy to break the hydration

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#4674] [General]
some libraries will change the document, for example [utteranc](https://utteranc.es/). And that will break the hydration.

I can fix this by include the library after hydration - do it inside `useEffect` hook.

```js
useEffect(() => {
  // import utterance
}, [])
```

But in StrictMode, the hook will execute twice. 

Can I change the hydrate node?

**Comment by lili21** [comment]
I already ran into 3 library that would insert element in the body event just include/import the library. which break the hydration.

I guess If we can just hydrate on a specify element, then we can prevent this problem.

**Comment by kiliman** [comment]
I've got a solution for the hydration issues entirely in userland.

https://github.com/kiliman/remix-hydration-fix
